File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int bricks1, int bricks2);
int height;

int main(void)
{
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    for (int i = 0; i < height; i++)
    {
        print_row(i + 1, i + 1, i + 1);
    }
}

void print_row(int spaces, int bricks1, int bricks2)
{
    for (int i = height; i > spaces; i--)
    {
        printf(" ");
    }
    for (int i = 0; i < bricks1; i++)
    {
        printf("#");
    }
    printf("  ");
    for (int i = 0; i < bricks2; i++)
    {
        printf("#");
    }
    printf("\n");
}

File: setimental-mario-more/mario.py
def print_row(spaces, bricks1, bricks2, height):
    print(" " * (height - spaces) + "#" * bricks1 + "  " + "#" * bricks2)

def main():
    while True:
        try:
            height = int(input("Height: "))
            if 1 <= height <= 8:
                break
        except ValueError:
            pass

    for i in range(1, height + 1):
        print_row(i, i, i, height)

if __name__ == "__main__":
    main()
